### PR TITLE
Added Sprout Glossary and Updated README

### LINKED ISSUE #12: Create MD file of Sprout concepts
We need a glossary to explain what this API does:

AC:

- start a docs folder
- include definitions for: threads, users, api keys, messages, files, tools, embeddings etc.
- include a link to these definitions in the readme

### PR DESCRIPTION
**Pull request by resourcefulmind**
This pull request introduces the `GLOSSARY.md` file to the Sprout project and updates the `README.md` to include a link to this new glossary. The purpose of these changes is to enhance the documentation of the project and provide easy access to definitions of key terms used in Sprout.

### Changes Made:

1. Added `GLOSSARY.md`:

- Created a new file docs/GLOSSARY.md.
- Included definitions for key terms such as Threads, Users, API Keys, Messages, Files, Tools, Embeddings, and Permissions.
- These terms are defined in the context of their use within the Sprout project, making it easier for new contributors and users to understand the project's terminology.

2. Updated `README.md`: 

- Inserted a link to the `GLOSSARY.md` in the `README.md` file.
- This link provides quick access to the glossary, aiding in the onboarding process of new project members and users.

### Reason for Changes:

The addition of a glossary is intended to improve the clarity and accessibility of the Sprout project's documentation. It provides a central place for definitions of key terms, which is especially beneficial for new contributors and users unfamiliar with the project's specific language. The update to the `README.md` ensures that this resource is easily discoverable.

### Additional Notes:

- The definitions in the glossary are a starting point and can be expanded or refined as the project evolves.
- Feedback and suggestions for additional terms or refinements are welcome.

closes #12

### CODE REVIEW

File: docs/GLOSSARY.MD [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+---
+
+## Threads
+
+In the context of Sprout, a thread refers to a sequence of messages or interactions that are linked or related to a single topic or user query. Threads in Sprout are crucial for maintaining the continuity of conversation or data processing. They allow the API to handle complex, multi-turn interactions by keeping track of the context.
+
+## Users
+
+Users are individuals or entities who interact with or utilize the Sprout API typically for generating or receiving AI-based text responses. In Sprout, users can be developers integrating the API into their applications, or end-users interacting with the final product. The user experience is central to Sprout's design and functionality.
```
Comment (GuiBibeau) [inline docs/GLOSSARY.MD:13]: There is a tiny nuance here: Users can also be assistants(agents). To be ready for greater autonomy in assistants, the permission system includes and applies to them.

This looks small on the surface but is key to how the system works.

**Review by GuiBibeau — COMMENTED** [review]
all is good a part from adding that users can be assistants as well.

File: docs/GLOSSARY.MD (hunk 1 above)
Comment (resourcefulmind) [reply docs/GLOSSARY.MD:13]: I agree. I will make the needed changes.

**Review by GuiBibeau — APPROVED** [review]
🔥 🚀

**Pull request merged**